### ISSUE TITLE
Prodive list_menu without radiolist

### ISSUE THREAD

**Issue by ghost** [autokey/autokey#518]
## Classification:

Enhancement

## Version

AutoKey version: 0.95.10
Used GUI (Gtk, Qt, or both): GTK
Installed via: apt
Linux Distribution: Ubuntu 20.04

## Summary

I would like to create a list dialogue without a radio buttons. Zenity provides an option to create these dialogues when "--radiolist" is **not** used as a parameter. Ideally, there would be an extra method such as dialog.list_menu_plain in order to avoid breaking changes for existing users of dialog.list_menu.

**Comment by josephj11** [comment]
TL;DR: so just use Zenity directly. :)

While we could expand our API to include more dialog manager options (we'd have to do it separately for each front end), it's pretty easy to use subprocess() to call out to the dialog manager of your choice or even to your own programs that use GTK or Qt directly to do anything you desire.

The way Zenity (which is what AutoKey uses for the GTK front end) works is that it returns somewhat structured text strings and several return statuses, so it's quite easy retrieve those into Python.

I'm a yad fan. It's a Zenity fork with even more options. I have successfully tested calling it from within AutoKey.

Here is a minimal example. You'd have to replace `subprocess.call()` with `subprocess.check_output()` or `subprocess.run()` to capture the output.

```
##andyad example script

import subprocess

def run_yad(text):
    q = '"'
    timeout = "--timeout=" + dsptime
    mytext = "--text=" + text
    subprocess.call([ "yad", "--title", name, '--on-top', '--width=300', 'height=250', '--center', '--button=gtk-ok:0', timeout, mytext])
   
dsptime = "5" ## yad timeout
name = "Andyad" ## Yad dialog title
message = "This is the message"
run_yad(message)
```

See [these](https://stackoverflow.com/questions/14078117/how-do-you-use-subprocess-check-output-in-python) other [examples](https://www.golinuxcloud.com/python-subprocess/).

Python keeps evolving and I am a Python beginner, so there may be newer, better ways of doing this, but they work.

Since you're calling out to the dialog manager, the same code will work with both AutoKey front ends because the external dialog manager doesn't rely on anything AutoKey is doing or provides for its functioning. This also allows you to develop and test your dialogs standalone outside of AutoKey where you have access to all your system tools and debugging aids.

**Comment by ghost** [comment]
Thank for your help. While your approach certainly works, as an AutoKey user and not hard core python programmer, I find it much easier to use existing methods. Since I wanted to have the return code and things like that, I ended up reimplementing things like run_zenity and DialogData - not sure if that is a good solution.

For anybody who wants a zenity list without radio buttons, here is the code:

```
def list_menu_plain(options, title="Choose a value", message="Choose a value", default=None, **kwargs):
    return run_zenity(
        title,
        ["--list", "--text", message, "--column", "Options"] + options,
        kwargs)
```

**Comment by josephj11** [comment]
Thanks for the example code.
